src/app/util/ClientMonitoringRegistrationTable.cpp

```diff
@@ -17,59 +17,71 @@
 
 #include "ClientMonitoringRegistrationTable.h"
 
-namespace chip {
+#include <lib/support/DefaultStorageKeyAllocator.h>
 
-/**********************************************************
- * Attributes Definition
- *********************************************************/
+namespace chip {
 
 /**********************************************************
  * ClientMonitoringRegistrationTable Implementation
  *********************************************************/
 
-ClientMonitoringRegistrationTable::ClientMonitoringRegistrationTable(FabricIndex fabricIndex)
-{
-    this->LoadFromStorage(fabricIndex);
-}
+ClientMonitoringRegistrationTable::ClientMonitoringRegistrationTable(PersistentStorageDelegate & storage) : mStorage(storage) {}
 
-void ClientMonitoringRegistrationTable::LoadFromStorage(FabricIndex fabricIndex)
+/**
+ * @brief Function loads a client registration entry from persistent storage for a single fabric
+ *
+ * @param fabricIndex fabric index to load from storage
+ * @return CHIP_ERROR
+ */
+CHIP_ERROR ClientMonitoringRegistrationTable::LoadFromStorage(FabricIndex fabricIndex)
 {
-    // TODO: Implement load from NVM logic
-}
+    uint8_t buffer[kRegStorageSize] = { 0 };
+    uint16_t size                   = sizeof(buffer);
 
-void ClientMonitoringRegistrationTable::SaveToStorage()
-{
-    // Store to NVM based of class attributes
-}
+    ReturnErrorOnFailure(
+        mStorage.SyncGetKeyValue(DefaultStorageKeyAllocator::ClientMonitoringTableEntry(fabricIndex).KeyName(), buffer, size));
 
-NodeId ClientMonitoringRegistrationTable::getClientNodeId()
-{
-    return mRegisteredClient.clientNodeId;
-}
+    TLV::TLVReader reader;
+    reader.Init(buffer, size);
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
 
-uint64_t ClientMonitoringRegistrationTable::getICid()
-{
-    return mRegisteredClient.ICid;
-}
+    ReturnErrorOnFailure(mRegisteredClient.Decode(reader));
 
-FabricIndex ClientMonitoringRegistrationTable::getFaricIndex()
-{
-    return mRegisteredClient.fabricIndex;
-}
+    mRegisteredClient.fabricIndex = fabricIndex;
 
-void ClientMonitoringRegistrationTable::setClientNodeId(NodeId clientNodeId)
-{
-    mRegisteredClient.clientNodeId = clientNodeId;
+    return CHIP_NO_ERROR;
 }
 
-void ClientMonitoringRegistrationTable::setICid(uint64_t ICid)
+/**
+ * @brief Function saves the mRegisteredClient attribute to persitant storage
+ *        To correctly persit an entry, the values must be stored in the structures attributes
+ *
+ * @return CHIP_ERROR
+ */
+CHIP_ERROR ClientMonitoringRegistrationTable::SaveToStorage()
 {
-    mRegisteredClient.ICid = ICid;
+    VerifyOrReturnError(mRegisteredClient.IsValid(), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
```suggestion
    VerifyOrReturnError(mRegisteredClient.IsValid(), CHIP_ERROR_INCORRECT_STATE);
```
INVALID_ARGUMENT is used to indicate that one of the function/method arguments is invalid and this method does not take arguments.